lib/CleantalkSP/SpbctWP/VulnerabilityAlarm/VulnerabilityAlarmService.php

```diff
@@ -411,4 +397,51 @@ private static function mapApiResults($api_res)
         }
         return $results;
     }
+
+    public static function sendReport()
+    {
+        global $spbc;
+
+        if ( $spbc->settings['spbc_key'] === '' ) {
+            throw new \Exception(__('Access key is empty.', 'security-malware-firewall'));
+        }
+
+        if ( $spbc->data['service_id'] === '' ) {
+            throw new \Exception(__('Service ID is empty. Please, do cloud synchronization.', 'security-malware-firewall'));
+        }
+
+        $vulnerabilities = [];
+
+        // 1) Collect vulnerable plugins research IDs
+        if (
+            isset($spbc->scan_plugins_info['names_vulnerable_plugins']) &&
+            is_array($spbc->scan_plugins_info['names_vulnerable_plugins']) &&
+            count($spbc->scan_plugins_info['names_vulnerable_plugins']) > 0
+        ) {
+            /** @var \CleantalkSP\SpbctWP\VulnerabilityAlarm\Dto\PluginReport $plugin_report */
+            foreach ( $spbc->scan_plugins_info['names_vulnerable_plugins'] as $plugin_report ) {
+                $vulnerabilities[] = $plugin_report->id;
+            }
+        }
+
+        // 2) Collect vulnerable themes research IDs
+        if (
+            isset($spbc->scan_themes_info['names_vulnerable_themes']) &&
+            is_array($spbc->scan_themes_info['names_vulnerable_themes']) &&
+            count($spbc->scan_themes_info['names_vulnerable_themes']) > 0
+        ) {
+            /** @var \CleantalkSP\SpbctWP\VulnerabilityAlarm\Dto\ThemeReport $theme_report */
+            foreach ( $spbc->scan_themes_info['names_vulnerable_themes'] as $theme_report ) {
+                $vulnerabilities[] = $theme_report->id;
+            }
+        }
+
+        // 3) Send found vulnerable IDs
+        if ( count($vulnerabilities) > 0 ) {
+            // @ToDo make only one API request after it will be able to receive array of IDs as the input parameter https://doboard.com/1/task/11753
+            foreach ( $vulnerabilities as $research_id ) {
+                $_send_result = API::methodReportVulnerabilityInUse($spbc->settings['spbc_key'], $spbc->data['service_id'], $research_id);
```

Comment:
Is this `_send_result` kept for further error handling?

Reply:
Yea